### ISSUE TITLE
Refactor code to return `error`s instead of using `os.Exit()`

### ISSUE THREAD

**Issue by FlorianLoch** [pepa65/sds011#1]
Hello there,
I would want to suggest to refactor some parts of the code in order to make using the project as a module in other projects easier/safer.

Most significantly, I would suggest to return `error`s instead of using `os.Exit()` in functions that are called by library users as error handling should be up to them and killing the whole application does not seem to be a nice thing to do ;)

If you like I will provide a PR - just wanted to check first ;)

**Comment by pepa65** [comment]
Yes, of course, that's a good idea. Thank you for working on this.

**Comment by pepa65** [comment]
I guess the `os.Exit` calls in `Sensor` need to return an error value too (and need checking when called). The one in SetDuty should probably just be a return.

**Comment by FlorianLoch** [comment]
Yes, these are the ones I am talking about ;)

Guess it shouldn't take me too long to adjust this, will try to do it later today 👍
